src/MonoTorrent/MonoTorrent.Client/Modes/Mode.cs

```diff
@@ -85,50 +86,50 @@ public void HandleMessage (PeerId id, PeerMessage message)
-                HandleLtChat (id, (LTChat) message);
-            else if (message is PeerExchangeMessage)
-                HandlePeerExchangeMessage (id, (PeerExchangeMessage) message);
-            else if (message is HandshakeMessage)
-                HandleHandshakeMessage (id, (HandshakeMessage) message);
-            else if (message is ExtensionMessage)
-                HandleGenericExtensionMessage (id, (ExtensionMessage) message);
+            if (message is HaveMessage have)
+                HandleHaveMessage (id, have);
+            else if (message is RequestMessage request)
+                HandleRequestMessage (id, request);
+            else if (message is PortMessage port)
+                HandlePortMessage (id, port);
+            else if (message is PieceMessage piece)
+                HandlePieceMessage (id, piece);
+            else if (message is NotInterestedMessage notinterested)
+                HandleNotInterested (id, notinterested);
+            else if (message is KeepAliveMessage keepalive)
+                HandleKeepAliveMessage (id, keepalive);
+            else if (message is InterestedMessage interested)
+                HandleInterestedMessage (id, interested);
+            else if (message is ChokeMessage choke)
+                HandleChokeMessage (id, choke);
+            else if (message is CancelMessage cancel)
+                HandleCancelMessage (id, cancel);
+            else if (message is BitfieldMessage bitfield)
+                HandleBitfieldMessage (id, bitfield);
+            else if (message is UnchokeMessage unchoke)
+                HandleUnchokeMessage (id, unchoke);
+            else if (message is HaveAllMessage haveall)
+                HandleHaveAllMessage (id, haveall);
+            else if (message is HaveNoneMessage havenone)
+                HandleHaveNoneMessage (id, havenone);
+            else if (message is RejectRequestMessage rejectrequest)
+                HandleRejectRequestMessage (id, rejectrequest);
+            else if (message is SuggestPieceMessage suggestpiece)
+                HandleSuggestedPieceMessage (id, suggestpiece);
+            else if (message is AllowedFastMessage allowedfast)
+                HandleAllowedFastMessage (id, allowedfast);
+            else if (message is ExtendedHandshakeMessage extendedhandshake)
+                HandleExtendedHandshakeMessage (id, extendedhandshake);
+            else if (message is LTMetadata metadata)
+                HandleLtMetadataMessage (id, metadata);
+            else if (message is LTChat chat)
+                HandleLtChat (id, chat);
+            else if (message is PeerExchangeMessage peerexchange)
+                HandlePeerExchangeMessage (id, peerexchange);
+            else if (message is HandshakeMessage handshake)
+                HandleHandshakeMessage (id, handshake);
+            else if (message is ExtensionMessage extension)
+                HandleGenericExtensionMessage (id, extension);
```

Comment:
Can you submit this as a separate PR. It's a very self-contained change and could merge all by itself :)

+1 on using the modern syntax here!